### PR TITLE
[SPARK-35425][BUILD] Pin jinja2 in `spark-rm/Dockerfile` and add as a required dependency in the release README.md

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

The following two things are done in this PR.

* Add note about Jinja2 as a required dependency for document build.
* Add Jinja2 dependency for the document build to `spark-rm/Dockerfile`

### Why are the changes needed?

SPARK-35375(#32509) confined the version of Jinja to <3.0.0.
So it's good to note about it in `docs/README.md` and add the dependency to `spark-rm/Dockerfile`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

I confimed that `make html` succeed under `python/docs` with the following command.
```
sudo pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsphinx numpydoc 'jinja2<3.0.0'
```

### CODE REVIEW

**Review by gengliangwang — APPROVED** [review]
LGTM

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, @sarutak .
In this case, we need to update `spark-rm/Dockerfile` together.
- https://github.com/apache/spark/blob/master/dev/create-release/spark-rm/Dockerfile#L43

**Comment by dongjoon-hyun** [comment]
cc @viirya

**Comment by sarutak** [comment]
@dongjoon-hyun Thanks. I didn't notice it.

File: dev/create-release/spark-rm/Dockerfile [hunk 1]
```diff
@@ -40,7 +40,9 @@ ARG APT_INSTALL="apt-get install --no-install-recommends -y"
 # TODO(SPARK-32407): Sphinx 3.1+ does not correctly index nested classes.
 #   See also https://github.com/sphinx-doc/sphinx/issues/7551.
 #   We should use the latest Sphinx version once this is fixed.
-ARG PIP_PKGS="sphinx==3.0.4 mkdocs==1.1.2 numpy==1.19.4 pydata_sphinx_theme==0.4.1 ipython==7.19.0 nbsphinx==0.8.0 numpydoc==1.1.0"
+# TODO(SPARK-35375): Jinja2 3.0.0+ causes error when building with Sphinx.
+#   See also https://issues.apache.org/jira/browse/SPARK-35375.
+ARG PIP_PKGS="sphinx==3.0.4 mkdocs==1.1.2 numpy==1.19.4 pydata_sphinx_theme==0.4.1 ipython==7.19.0 nbsphinx==0.8.0 numpydoc==1.1.0 jinja2==2.11.3"
```
Comment (viirya) [inline dev/create-release/spark-rm/Dockerfile:45]: Why not also `jinja2<3.0.0`?

Comment (sarutak) [reply dev/create-release/spark-rm/Dockerfile:45]: Other packages are also fixed their version in this file, so should we fix the version for Jinja2 in this file too?

Comment (sarutak) [reply dev/create-release/spark-rm/Dockerfile:45]: Other dependencies are also fixed their version in this file, so should we fix for Jinja2 too?

Comment (viirya) [reply dev/create-release/spark-rm/Dockerfile:45]: Oh okay. I'm just curious why it is not as same as the other place. If we know the version works, then it's fine.

**Review by viirya — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you so much.

@sarutak . It seems that we need this fix on branch-3.1/3.0 too, right?

**Comment by sarutak** [comment]
@dongjoon-hyun #32509 was merged to `master` and `branch-3.1`. But I'll check whether we need to merge this to `branch-3.0` too.

**Comment by dongjoon-hyun** [comment]
Oh, what I was focused was `Dockerfile` change.

**Comment by sarutak** [comment]
> Oh, what I was focused was Dockerfile change.
Yes, I understand. I'll check that whether the change for the Dockerfile is necessary for `branch-3.0` though #32509 was not necessary.

**Comment by sarutak** [comment]
@dongjoon-hyun 

The Dockerfile in `branch-3.0` installs the following packages and without the version of Jinja2 specified but `make html` under `python/doc` succeeds. Maybe whether the version of Jinja2 affects or not is subject to the version of `Sphinx`.

* sphinx==2.3.1
* mkdocs==1.0.4
* numpy==1.18.1

But it might be pin the version of Jinja2 just in case for `branch-3.0` too. What do you think?

**Comment by dongjoon-hyun** [comment]
Thank you for investigation. +1 for pinning idea.

**Comment by sarutak** [comment]
I'll open abackport PR for `branch-3.0` because this change should not be merged to it straightly.

**Comment by sarutak** [comment]
Merged to `master`. Thanks all!.
I'll open backport PRs for `branch-3.1` and `branch-3.0`.

**Pull request closed without merging**